commons/src/main/java/io/aiven/kafka/connect/common/source/task/TaskAssignment.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.source.task;
+
+/**
+ * TaskAssignment is a common interface which allows source connectors to determine what method of distributing tasks
+ * amongst source connectors in distributed mode.
+ */
+public interface TaskAssignment {
+
+    boolean isPartOfTask(String valueToBeEvaluated);
+
+    void configureTask(int taskId, int maxTasks, String expectedFormat);
```

Comment:
This method initializes the TaskAssignment for a Task (likely one per task?)  It might be best just to use a signature like:

```
void initialize(Config taskConfig); 
```

Reply:
I was thinking about this, the reason I have named it configureTask, is for rebalancing connectors, we can re-configure a task assignment on the fly and not require a restart.

Last night I was thinking(when I was lying in bed awake :( ) a initializing the details in the constructor and renaming the configure to reconfigure might be better?
WDYT?

Reply:
First of all the naming right now seems to be ambiguous, we are not configuring a task but an assignment, right? 
Second thing is that I'm not sure if we really want to have an assignment per task + taskId as a field vs passing taskId when checking if specific partition is assigned to it.

Reply:
I like the idea of renaming these strategies, and having the taskId entered as part of it.

I think we just need a reconfigure which will reset the maxTasks when scaling, and this would work really well.